src/coreclr/vm/jitinterface.cpp

```diff
@@ -12021,10 +12025,17 @@ bool CEEInfo::getReadonlyStaticFieldValue(CORINFO_FIELD_HANDLE fieldHnd, uint8_t
         }
         else
         {
-            void* fldAddr = field->GetCurrentStaticAddress();
-            _ASSERTE(fldAddr != nullptr);
-            memcpy(buffer, fldAddr, bufferSize);
-            result = true;
+            // Either RVA, primitive or struct (or even part of that struct)
```

Comment:
Is it guaranteed/specced that if an RVA field is marked `readonly`, the backing storage for it cannot be not mutated? Does this need an additional check that the RVA is pointing into a read-only section?

Reply:
Ah, good point, forgot about C++CLI/CX - any idea how to check that?

Reply:
I don't know beyond using `VirtualQuery`. I can also see there is `PEDecoder::RvaToSection`, but not sure how to obtain that.

I doubt that C++/CLI would ever create RVA fields with this kind of disparity, so this is probably more of a theoretical issue. Maybe we don't need to do anything. @jkotas ?

Reply:
> if an RVA field is marked readonly, the backing storage for it cannot be not mutated

I think it is fine assumption to make.